app/models/concerns/vacancy_important_date_validations.rb

```diff
@@ -12,6 +12,8 @@ module VacancyImportantDateValidations
     validate :starts_on_must_not_be_before_today, if: :starts_on?
     validate :starts_on_must_not_be_before_publish_on, if: :starts_on?
     validate :starts_on_must_not_be_before_expires_on, if: :starts_on?
+
+    validate :starts_on_and_starts_asap_must_not_be_both_given, if: :starts_on? && :starts_asap?
```

Comment:
`validate :starts_on_and_starts_asap_must_not_both_be_present` would read better I think. Also, you don't need to add the if part here when that's identical to what the validation is doing